source/common/common/assert.h

```diff
@@ -46,23 +47,24 @@ ActionRegistrationPtr setDebugAssertionFailureRecordAction(const std::function<v
  * @param action The action to take when an envoy bug fails.
  * @return A registration object. The registration is removed when the object is destructed.
  */
-ActionRegistrationPtr setEnvoyBugFailureRecordAction(const std::function<void()>& action);
+ActionRegistrationPtr
+addEnvoyBugFailureRecordAction(const std::function<void(const char* location)>& action);
 
 /**
  * Invokes the action set by setDebugAssertionFailureRecordAction, or does nothing if
  * no action has been set.
  *
  * This should only be called by ASSERT macros in this file.
  */
-void invokeDebugAssertionFailureRecordActionForAssertMacroUseOnly();
+void invokeDebugAssertionFailureRecordActionForAssertMacroUseOnly(const char* location);
```

Comment:
nit: document location refers to ASSERT location (or something similar)

Reply:
Done.